include/nbl/video/IGPUCommandPool.h

```diff
@@ -4,38 +4,835 @@
+{
+public:
+    CBindIndexBufferCmd(core::smart_refctd_ptr<const video::IGPUBuffer>&& indexBuffer) : m_indexBuffer(std::move(indexBuffer)) {}
+
+private:
+    core::smart_refctd_ptr<const video::IGPUBuffer> m_indexBuffer;
+};
+
+class IGPUCommandPool::CDrawIndirectCmd : public IGPUCommandPool::IFixedSizeCommand<CDrawIndirectCmd>
+{
+public:
+    CDrawIndirectCmd(core::smart_refctd_ptr<const video::IGPUBuffer>&& buffer) : m_buffer(std::move(buffer)) {}
+
+private:
+    core::smart_refctd_ptr<const IGPUBuffer> m_buffer;
+};
+
+class IGPUCommandPool::CDrawIndexedIndirectCmd : public IGPUCommandPool::IFixedSizeCommand<CDrawIndexedIndirectCmd>
+{
+public:
+    CDrawIndexedIndirectCmd(core::smart_refctd_ptr<const video::IGPUBuffer>&& buffer) : m_buffer(std::move(buffer)) {}
+
+private:
+    core::smart_refctd_ptr<const IGPUBuffer> m_buffer;
+};
+
+class IGPUCommandPool::CDrawIndirectCountCmd : public IGPUCommandPool::IFixedSizeCommand<CDrawIndirectCountCmd>
+{
+public:
+    CDrawIndirectCountCmd(core::smart_refctd_ptr<const video::IGPUBuffer>&& buffer, core::smart_refctd_ptr<const video::IGPUBuffer>&& countBuffer)
+        : m_buffer(std::move(buffer)) , m_countBuffer(std::move(countBuffer))
+    {}
+
+private:
+    core::smart_refctd_ptr<const IGPUBuffer> m_buffer;
+    core::smart_refctd_ptr<const IGPUBuffer> m_countBuffer;
+};
+
+class IGPUCommandPool::CDrawIndexedIndirectCountCmd : public IGPUCommandPool::IFixedSizeCommand<CDrawIndexedIndirectCountCmd>
+{
+public:
+    CDrawIndexedIndirectCountCmd(core::smart_refctd_ptr<const video::IGPUBuffer>&& buffer, core::smart_refctd_ptr<const video::IGPUBuffer>&& countBuffer)
+        : m_buffer(std::move(buffer)), m_countBuffer(std::move(countBuffer))
+    {}
+
+private:
+    core::smart_refctd_ptr<const IGPUBuffer> m_buffer;
+    core::smart_refctd_ptr<const IGPUBuffer> m_countBuffer;
+};
+
+class IGPUCommandPool::CBeginRenderPassCmd : public IGPUCommandPool::IFixedSizeCommand<CBeginRenderPassCmd>
+{
+public:
+    CBeginRenderPassCmd(core::smart_refctd_ptr<const video::IGPURenderpass>&& renderpass, core::smart_refctd_ptr<const video::IGPUFramebuffer>&& framebuffer)
+        : m_renderpass(std::move(renderpass)), m_framebuffer(std::move(framebuffer))
+    {}
+
+private:
+    core::smart_refctd_ptr<const video::IGPURenderpass> m_renderpass;
+    core::smart_refctd_ptr<const video::IGPUFramebuffer> m_framebuffer;
+};
+
+class IGPUCommandPool::CPipelineBarrierCmd : public IGPUCommandPool::ICommand
+{
+public:
+    CPipelineBarrierCmd(const uint32_t bufferCount, const core::smart_refctd_ptr<const IGPUBuffer>* buffers, const uint32_t imageCount, const core::smart_refctd_ptr<const IGPUImage>* images)
+        : ICommand(calc_size(bufferCount, buffers, imageCount, images)), m_resourceCount(bufferCount+imageCount)
+    {
+        m_barrierResources = reinterpret_cast<core::smart_refctd_ptr<const core::IReferenceCounted>*>(reinterpret_cast<uint8_t*>(this) + sizeof(CPipelineBarrierCmd));
```

Comment:
I realized after the change, that this (and for other all variable-length commands) is better/more succinctly written as:
```C++
m_barrierResources = this + 1ull;
```
so I will make this change before merge.

Reply:
Resolved.

Reply:
not pushed yet?

Reply:
Yes, pushed now.